contracts/proxies/CurrencyConvertor.sol

```diff
@@ -58,6 +58,23 @@ contract CurrencyConvertor {
 
   // ============ State-Changing Functions ============
 
+      /**
+     * @notice Sets the maximum allowance on the L2 contract. Must be called at least once
+     *  before deposits can be made.
+     * @dev Cannot be run in the constructor due to technical restrictions in Solidity.
```

Comment:
Why can't this be run in the constructor? You can make external calls to contracts within a constructor: https://ethereum.stackexchange.com/a/63970